### PR TITLE
Support additional JDBC properties for a DataSource

### PR DESCRIPTION
**Pull request by gastaldi**
Some environments use these properties for additional metadata (eg. in Google CloudSQL)

Closes #6634

### CODE REVIEW

**Review by loicmathieu — APPROVED** [review]
LGTM

File: extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceJdbcRuntimeConfig.java [hunk 1]
```diff
@@ -103,4 +104,11 @@
      */
     @ConfigItem(defaultValue = "true")
     public boolean poolingEnabled = true;
+
+    /**
+     * Other unspecified properties to be passed into the JDBC driver when creating new connections
```
Comment (gsmet) [inline extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceJdbcRuntimeConfig.java:109]:
```suggestion
     * Other unspecified properties to be passed to the JDBC driver when creating new connections.
```

**Review by gsmet — DISMISSED** [review]
Looks good to me except for a very minor nitpick.

**Comment by Sanne** [comment]
LGTM, thanks!

**Comment by binojvr** [comment]
Any update on when this will be available

**Comment by gastaldi** [comment]
It's available in 1.9.0.Final (to be announced soon)

**Pull request merged**